src/models/hooks/activityReport.js

```diff
@@ -155,6 +155,78 @@ const propogateSubmissionStatus = async (sequelize, instance, options) => {
   }
 };
 
+const determineObjectiveStatus = async (activityReportId, sequelize) => {
+  // Get all AR Objective Id's.
+  const objectives = await sequelize.models.ActivityReportObjective.findAll(
+    {
+      attributes: [
+        'id',
+        'objectiveId',
+      ],
+      where: { activityReportId },
+    },
+  );
+  const objectiveIds = objectives.map((o) => o.objectiveId);
+
+  // Get all the reports that use the objectives
+  const allObjectiveReports = await sequelize.models.ActivityReport.findAll({
+    include: [
+      {
+        model: sequelize.models.ActivityReportObjective,
+        as: 'activityReportObjectives',
+        where: {
+          objectiveId: objectiveIds,
+        },
+        include: [{
+          model: sequelize.models.Objective,
+          as: 'objective',
+        }],
+      },
+    ],
+  });
+
+  // Get Approved Reports that might set the status.
+  const approvedReports = allObjectiveReports.filter(
+    (a) => a.calculatedStatus === REPORT_STATUSES.APPROVED,
+  );
+
+  // Check if there aren't any approved reports to set the status.
+  let updateObjectivePromises;
+  if (!approvedReports.length) {
+    // Get Objective Id's to reset.
+    const currentReport = allObjectiveReports.find((r) => r.id === activityReportId);
+    const objectiveIdsToReset = currentReport.activityReportObjectives.map((a) => a.objectiveId);
+    updateObjectivePromises = [
+      sequelize.models.Objective.update({
+        status: 'Not Started',
+      }, {
+        where: { id: objectiveIdsToReset },
+        individualHooks: true,
+      })];
```

Comment:
Do we want to automatically set them to Not Started? What if they were created as "In Progress" on the RTR or something?

Reply:
So a user wouldn't be able to set an existing objective to in progress on the RTR? Or if they did, it would be reverted as soon as they used on an activity report (regardless of what the status of the objective on that report is) until the report is approved?

Reply:
I would think once it's used on an AR it enters a different state that relies on an approved AR. But this raises the question of what purpose does setting an objective status from the RTR serve. It would only be used before it has been placed on an AR. Ill double check with @GarrettEHill  but it might be a question for ACF.

Reply:
Objectives are created as "Not Started" - why not leave it alone unless an approved report says so? Are we catching a certain case by just setting them all to "Not Started" here? Am I missing something?

Reply:
I guess it goes to the point of the latest approved report being a source of truth.

I could go either way.